SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -11,9 +11,70 @@
 #include "EngineClasses/SpatialPackageMapClient.h"
 #include "SpatialGDKSettings.h"
 #include "SpatialConstants.h"
+#include "UObject/UObjectIterator.h"
 
 DEFINE_LOG_CATEGORY(LogInterestFactory);
 
+namespace
+{
+static TMap<UClass*, float> ClientInterestDistances;
+void GatherClientInterestDistancesOnce()
+{
+	if (ClientInterestDistances.Num() > 0)
+	{
+		return;
+	}
+
+	const AActor* DefaultActor = Cast<AActor>(AActor::StaticClass()->GetDefaultObject());
+	const float DefaultDistance = DefaultActor->ClientInterestDistance;
+
+	// Gather ClientInterestDistance settings, and add any larger than the default radius to a list for processing.
+	TMap<UClass*, float> DiscoveredInterestDistances;
+	for (TObjectIterator<UClass> It; It; ++It)
+	{
+		if (!It->IsChildOf<AActor>())
+		{
+			continue;
+		}
+
+		const AActor* IteratedDefaultActor = Cast<AActor>(It->GetDefaultObject());
+		if (IteratedDefaultActor->ClientInterestDistance > DefaultDistance)
```

Comment:
I think we can also ignore classes here that have been tagged with the spatial flag `ServerOnly`